hnn_core/network.py

```diff
@@ -374,20 +379,26 @@ def add_evoked_drive(self, name, *, mu, sigma, numspikes,
         if not self._legacy_mode:
             _check_drive_parameter_values('evoked', sigma=sigma,
                                           numspikes=numspikes)
-
         drive = _NetworkDrive()
         drive['type'] = 'evoked'
         if name == 'extgauss':
             drive['type'] = 'gaussian'  # XXX needed to pass legacy tests!
-        drive['cell_specific'] = True
+        if numsources is None:
+            numsources = self.n_cells
+            drive['cell_specific'] = True
+        else:
+            _validate_type(numsources, types=int)
+            _check_option('numsources', numsources, allowed_values=[1])
+            drive['cell_specific'] = False
```

Comment:
Is `'cell_specific'` still going to be needed?

Reply:
> Is `'cell_specific'` still going to be needed?

Honestly, I'm not sure yet. We'll still need a parameter that flags one-to-one vs all-to-all drive connectivity, but it probably doesn't need to be an item within the `drive` dict.

Reply:
Pretty sure it will still be necessary when `n_drive_cells=None`